### PR TITLE
[local] Add ipc/pid fields

### PR DESCRIPTION
**Pull request by SoManyHs**
Also refactored metadata logic

Addresses #810 

----
Enter `[N/A]` in the box, if an item is not applicable to your change.

**Testing**  
- [x] Unit tests passed
- [x] Integration tests passed
- [x] Unit tests added for new functionality
- [x] Listed manual checks and their outputs in the comments ([example](https://github.com/aws/amazon-ecs-cli/pull/750#issuecomment-472623042))
- [x] Link to issue or PR for the integration tests: 

**Documentation**  
- [ ] Contacted our doc writer
- [ ] Updated our README
----

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: ecs-cli/modules/cli/local/localproject/project.go [hunk 1]
```diff
@@ -121,7 +129,23 @@ func (p *localProject) ReadTaskDefinition() error {
 	return nil
 }
 
+var defaultInputExists = func() bool {
+	_, err := os.Stat(LocalInFileName)
```
Comment (efekarakus) [inline ecs-cli/modules/cli/local/localproject/project.go]: nit: to be more concise
```go
_, err := os.Stat(LocalInFileName); err != nil {
   return false
}
return true
```

File: ecs-cli/modules/cli/local/converter/converter.go [hunk 2]
```diff
@@ -39,35 +39,41 @@ type LinuxParams struct {
...
-// See https://github.com/aws/amazon-ecs-cli/issues/797
-const SecretLabelPrefix = "ecs-local.secret"
+type LocalCreateMetadata struct {
+	InputType string
+	Value     string
+}
+
+type CommonContainerValues struct {
+	NetworkMode string
+	Ipc         string
```
[the remark this answers is not in the document]
Comment (SoManyHs) [reply ecs-cli/modules/cli/local/converter/converter.go]: That's fine but I was trying to keep this consistent with the compose v3 types.

File: ecs-cli/modules/cli/local/converter/converter.go [hunk 3]
```diff
@@ -91,10 +98,49 @@ func ConvertToDockerCompose(taskDefinition *ecs.TaskDefinition, localTaskType, l
 	return data, nil
 }
 
-// TODO convert top level volumes
-// TODO convert top level Neworks
+func createComposeServices(taskDefinition *ecs.TaskDefinition, metadata *LocalCreateMetadata) ([]composeV3.ServiceConfig, error) {
+	networkMode := aws.StringValue(taskDefinition.NetworkMode)
+	// TODO add log info message to indicate networkMode should be empty for Windows containers
+	if networkMode == ecs.NetworkModeAwsvpc {
```
[the remark this answers is not in the document]
Comment (SoManyHs) [reply ecs-cli/modules/cli/local/converter/converter.go]: hence the TODO message.

File: ecs-cli/modules/cli/local/converter/converter.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (SoManyHs) [reply ecs-cli/modules/cli/local/converter/converter.go]: As mentioned [here](https://github.com/aws/amazon-ecs-cli/pull/815#discussion_r298344468), this is consistent with what the types are called in the docker/cli lib: https://github.com/aws/amazon-ecs-cli/blob/master/ecs-cli/vendor/github.com/docker/cli/cli/compose/types/types.go#L123

**Comment by SoManyHs** [comment]
Manual test:

```
~/ecs/ecs-cli/testing/ecs_local$ ecs-cli -v
ecs-cli version 1.14.1 (*270e9a4)
```

Task definition:
```
~/ecs/ecs-cli/testing/ecs_local$ cat task-definition.json 

{
   "containerDefinitions": [
      {
         "image": "nginx",
         "name": "web"
      },
      {
         "command": [
            "/bin/sh -c \"echo '<html> <head> <title>Amazon ECS Sample App</title> <style>body {margin-top: 40px; background-color: #333;} </style> </head><body> <div style=color:white;text-align:center> <h1>Amazon ECS Sample App</h1> <h2>Congratulations!</h2> <p>Your application is now running on a container in Amazon ECS.</p> </div></body></html>' >  /usr/local/apache2/htdocs/index.html && httpd-foreground\""
         ],
         "entryPoint": [
            "sh",
            "-c"
         ],
         "essential": true,
         "image": "httpd:2.4",
         "logConfiguration": {
            "logDriver": "awslogs",
            "options": {
               "awslogs-group" : "/ecs/fargate-task-definition",
               "awslogs-region": "us-east-1",
               "awslogs-stream-prefix": "ecs"
            }
         },
	 "secrets": [{
		 "name": "MY_SECRET",
		 "valueFrom": "arn:aws:secretsmanager:us-west-2:794715269151:secret:MY_SECRET"
	 }],
         "name": "sample-fargate-app",
         "portMappings": [
            {
               "containerPort": 80,
               "hostPort": 80,
               "protocol": "tcp"
            }
         ]
      }
   ],
   "cpu": "256",
   "executionRoleArn": "arn:aws:iam::012345678910:role/ecsTaskExecutionRole",
   "family": "fargate-task-definition",
   "memory": "512",
   "networkMode": "bridge"
}

```

Running local create with no flags (uses default input and output files):
```
~/ecs/ecs-cli/testing/ecs_local$ ecs-cli local create
INFO[0000] If you are running windows containers, make sure network mode is not set.  networkMode=bridge
Successfully wrote docker-compose.local.yml

~/ecs/ecs-cli/testing/ecs_local$ cat docker-compose.local.yml 
version: "3.0"
services:
  sample-fargate-app:
    command:
    - '/bin/sh -c "echo ''<html> <head> <title>Amazon ECS Sample App</title> <style>body
      {margin-top: 40px; background-color: #333;} </style> </head><body> <div style=color:white;text-align:center>
      <h1>Amazon ECS Sample App</h1> <h2>Congratulations!</h2> <p>Your application
      is now running on a container in Amazon ECS.</p> </div></body></html>'' >  /usr/local/apache2/htdocs/index.html
      && httpd-foreground"'
    entrypoint:
    - sh
    - -c
    environment:
      MY_SECRET: ${sample-fargate-app_MY_SECRET}
    image: httpd:2.4
    labels:
      ecs-local.secret.MY_SECRET: arn:aws:secretsmanager:us-west-2:794715269151:secret:MY_SECRET
      ecs-local.task-definition-input.type: local
      ecs-local.task-definition-input.value: task-definition.json
    logging:
      driver: awslogs
      options:
        awslogs-group: /ecs/fargate-task-definition
        awslogs-region: us-east-1
        awslogs-stream-prefix: ecs
    network_mode: bridge
    networks:
      ecs-local-network: null
    ports:
    - target: 80
      published: 80
      protocol: tcp
  web:
    image: nginx
    labels:
      ecs-local.task-definition-input.type: local
      **ecs-local.task-definition-input.value: task-definition.json**
    network_mode: bridge
    networks:
      ecs-local-network: null
networks:
  ecs-local-network:
    external: true
```

**Comment by SoManyHs** [comment]
Note: With the above docker compose file, `up` fails:
```
~/ecs/ecs-cli/testing/ecs_local$ docker-compose up
ERROR: 'network_mode' and 'networks' cannot be combined
```

Since we're using the local container endpoints network, we will not populate network mode in services at all.

File: ecs-cli/modules/cli/local/converter/converter.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (SoManyHs) [reply ecs-cli/modules/cli/local/converter/converter.go]: As per meatspace discussion we're going to rip this out.

**Comment by SoManyHs** [comment]
Using basically the same docker compose file as above (changing to 3.2 for reasons):

```
~/ecs/ecs-cli/testing/ecs_local$ cat docker-compose.yml 
version: "3.2"
services:
  sample-fargate-app:
    command:
    - '/bin/sh -c "echo ''<html> <head> <title>Amazon ECS Sample App</title> <style>body
      {margin-top: 40px; background-color: #333;} </style> </head><body> <div style=color:white;text-align:center>
      <h1>Amazon ECS Sample App</h1> <h2>Congratulations!</h2> <p>Your application
      is now running on a container in Amazon ECS.</p> </div></body></html>'' >  /usr/local/apache2/htdocs/index.html
      && httpd-foreground"'
    entrypoint:
    - sh
    - -c
    environment:
      MY_SECRET: ${sample-fargate-app_MY_SECRET}
    image: httpd:2.4
    ipc: task
    labels:
      ecs-local.secret.MY_SECRET: arn:aws:secretsmanager:us-west-2:794715269151:secret:MY_SECRET
      ecs-local.task-definition-input.type: local
      ecs-local.task-definition-input.value: task-definition.json
    logging:
      driver: awslogs
      options:
        awslogs-group: /ecs/fargate-task-definition
        awslogs-region: us-east-1
        awslogs-stream-prefix: ecs
    networks:
      ecs-local-network: null
    pid: host
    ports:
    - target: 80
      published: 80
      protocol: tcp
  web:
    image: nginx
    ipc: task
    labels:
      ecs-local.task-definition-input.type: local
      ecs-local.task-definition-input.value: task-definition.json
    networks:
      ecs-local-network: null
    pid: host
networks:
  ecs-local-network:
    external: true
```

when running up:

```
~/ecs/ecs-cli/testing/ecs_local$ docker-compose up
Creating ecs_local_web_1 ... 
Creating ecs_local_sample-fargate-app_1 ... error

Creating ecs_local_web_1                ... error
driver

ERROR: for ecs_local_web_1  Cannot start service web: linux spec namespaces: Invalid IPC mode: task

ERROR: for sample-fargate-app  Cannot create container for service sample-fargate-app: unknown log opt 'awslogs-stream-prefix' for awslogs log driver

ERROR: for web  Cannot start service web: linux spec namespaces: Invalid IPC mode: task
ERROR: Encountered errors while bringing up the project.
```

ECS IPC mode values: host, task, none
DC IPC values: https://docs.docker.com/engine/reference/run/#ipc-settings---ipc (host or none only compatible ones)

ECS PID mode values: host or task
DC PID values: host https://docs.docker.com/compose/compose-file/#pid

**Pull request merged**